deidentification/anonymizer.py

```diff
@@ -16,82 +14,142 @@
+    return tags_to_keep
 
 
 def anonymize(dicom_in, dicom_out,
-              tags_to_keep=tags_to_keep,
-              forced_values=forced_values):
-    """
-    Configures the Anonymizer and runs it on DICOM files.
+              tags_to_keep=None,
+              forced_values=None,
+              config_profile=None,
+              anonymous=False):
+    """Configures the Anonymizer and runs it on DICOM files.
+    
+    Parameters
+    ----------
+    dicom_in : str
+    dicom_out : str
+    tags_to_keep : list, optional
+    forced_values : dict, optional
+    config_profile : str, optional
+    anonymous : bool, optional
     """
     if not os.path.exists(dicom_in):
-        print("The DICOM input does not exist.")
-        return
+        raise DeidentificationError('The DICOM input does not exists.')
+    if not os.path.isfile(dicom_in) and not os.path.isdir(dicom_in):
+        raise DeidentificationError('The DICOM input file type is not handled by this tool.')
+    if not os.path.isfile(dicom_in) and os.path.isfile(dicom_out):
+        raise DeidentificationError('The DICOM out could not be a file if DICOM in is not a file.')
+    if config_profile:
+        if tags_to_keep:
+            raise DeidentificationError('Both tags_to_keep and config_profile have been specified.')
+        tags_to_keep = load_config_profile(config_profile)
 
+    # Handle archives
     is_dicom_in_archive = is_archive(dicom_in)
     is_dicom_out_archive = is_archive(dicom_out)
     if is_dicom_in_archive:
         wip_dicom_in = mkdtemp()
-        unpack(dicom_in, wip_dicom_in)
+        try:
+            unpack(dicom_in, wip_dicom_in)
+        except Exception:
+            raise DeidentificationError('Unpacking compressed file failed.')
     else:
         wip_dicom_in = os.path.abspath(dicom_in)
     if is_dicom_out_archive:
         wip_dicom_out = mkdtemp()
     else:
         wip_dicom_out = os.path.abspath(dicom_out)
 
+    # Launch deidentification
     if os.path.isfile(wip_dicom_in):
         anonymize_file(wip_dicom_in, wip_dicom_out,
-                       tags_to_keep, forced_values)
+                       tags_to_keep, forced_values,
+                       anonymous=anonymous)
 
     elif os.path.isdir(wip_dicom_in):
-        if os.path.isfile(wip_dicom_out):
-            print("Since the input is a directory, an output directory is expected.")
-            return
-        
         for root, dirs, files in os.walk(wip_dicom_in):
             folder_out = root.replace(wip_dicom_in, wip_dicom_out)
             for name in files:
                 current_file = os.path.join(root, name)
                 anonymize_file(current_file, folder_out,
-                               tags_to_keep, forced_values)
+                               tags_to_keep, forced_values,
+                               anonymous=anonymous)
                 
-    else:
-        print("The input file type is not handled by this tool.")
-
     if is_dicom_in_archive:
```

Comment:
Deleting of `wip_dicom_in` is very important, what happens if, for example, an error is raised inside `anonymize_file`, the folder thtat contains raw data is not removed ? I think we must better handle this part. Same for `wip_dicom_out`.